biogrid/libs/biogrid-simulator/src/lib/bioenergy-source/solar-panel.ts

```diff
@@ -71,8 +71,8 @@ export class SolarPanel extends EnergySource {
    * This method returns the current power that can be generated by
    * the solar panel at that given time
    */
-  async getEnergyInJoules(): Promise<Power> {
-    return await this.getPowerAmount(this.date);
+  async getEnergyInJoules(date?: Date): Promise<Power> {
+    return await this.getPowerAmount(this.date || date);
```

Comment:
what if the date param is empty/null? 
Is this.date supposed to provide a default value fallback? If that's the case - do you mean `date || this.date` instead?

Reply:
Yeah, I fixed this is in a future PR after figuring that all out

Reply:
In PR #85

Reply:
Can we apply the fix here? Doesn't make sense to push a change that is incorrect even if it is immediately solved in the future.